frontend/src/app/visualization/ngram/ngram.component.spec.ts

```diff
@@ -38,32 +55,54 @@ describe('NgramComponent', () => {
     component.visualizedField = {name: 'speech'} as any;
     component.asTable = false;
     component.palette = ['yellow', 'blue'];
+    spyOn(component.ngramParameters, 'getCurrentRouterState').and.returnValue(cacheKey);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should initialize ngramParameters with default values', () => {
+    expect(component.ngramParameters.state$.value).toEqual(defaultSettings);
+  });
+
+  it('should not abort tasks when `onParameterChange` is triggered during initialization', () => {
+    spyOn(component.stopPolling$, 'next');
+    component.onParameterChange('size', 2);
+    expect(component.stopPolling$.next).not.toHaveBeenCalled();
+  })
+
```

Comment:
On initialization, all values in the template will be adjusted to the values reflected in the routing parameters, which will trigger the `onParameterChange` method, hence the extra `if` condition in there to see if the parameter *has* actually changed.